aiidalab_ispg/spectrum_analysis.py

```diff
@@ -224,3 +241,389 @@ def _observe_disabled(self, change):
+
+    def handle_slider_change(self, change):
+        """Redraw spectra when user changes quantum yield via slider
+
+        :param change: The change object containing the old and new values of the yield slider.
+        :type change: dict
+        """
+        self._update_j_plot(plot_type=self.flux_toggle.value, quantumY=change.new)
+
+    # **********************************************************************
+
+    # **********************************************************************
+    def _update_j_plot(self, plot_type: str, quantumY):
+        """
+        Update the J plot based on the given plot type and quantum yield
+
+        :param plot_type: The flux of plot to generate. Can be "LOW", "MED", or "HIGH".
+        :type plot_type: str
+        :param quantumY: The quantum yield value to use in the calculation.
+        :type quantumY: float
+        :return: A tuple containing the J values and wavelengths used in the plot.
+        :rtype: tuple
+        """
+
+        if self.spectrum_data is None:
+            return
+        if plot_type == "LOW":
+            j_values = self.calculation(1, quantum_yield=quantumY)
+            wavelengths = self.flux_data[0]
+            self.plot_line(wavelengths, j_values, label="label")
+            self.add_log_axis(wavelengths, 1, label="log")
+        elif plot_type == "MED":
+            j_values = self.calculation(2, quantum_yield=quantumY)
+            wavelengths = self.flux_data[0]
+            self.plot_line(wavelengths, j_values, label="label")
+            self.add_log_axis(wavelengths, 2, label="log")
+
+        elif plot_type == "HIGH":
+            j_values = self.calculation(3, quantum_yield=quantumY)
+            wavelengths = self.flux_data[0]
+            self.plot_line(wavelengths, j_values, label="label")
+            self.add_log_axis(wavelengths, 3, label="log")
+
+        else:
+            raise ValueError(f"Unexpected value for toggle: {plot_type}")
+
+        self.number.value = f"{np.format_float_scientific(np.trapz(j_values, dx=1),3)}"
+        return j_values, wavelengths
+
+    # **********************************************************************
+
+    # **********************************************************************
+    def reset(self):
+        """
+        Reset the figure and its associated widgets to their default values.
+        """
+        with self.hold_trait_notifications():
+            self.disabled = True
+            self.figure.clean()
+            self.flux_toggle.value = "HIGH"
+            self.yield_slider.value = 1
+            self.number.value = "None"
+
+    @traitlets.observe("disabled")
+    def _observe_disabled(self, change):
+        """
+        Observe changes to the "disabled" trait and update the state of the flux_toggle and yield_slider widgets accordingly.
+
+        :param change: A dictionary containing information about the change to the "disabled" trait.
+        """
```

Comment:
Because the observer pattern is so common in all the widgets, here I'd say the docstring is a bit redundant, especially since it is a private method anyway. The function name and the decorator already make things clear enough.